### ISSUE TITLE
Restic fatal error when backing up a zfs snapshot

### ISSUE THREAD

**Issue by creaky** [restic/restic#3839]
Output of `restic version`
--------------------------
restic 0.13.1 compiled with go1.18 on solaris/amd64

Operating System is SmartOS joyent_20220714T060459Z i86pc i386 i86pc

How did you run restic exactly?
-------------------------------

export B2_ACCOUNT_ID='redacted'
export B2_ACCOUNT_KEY='redacted'
export RESTIC_REPOSITORY='redacted'
export RESTIC_PASSWORD='redacted'

/opt/local/bin/restic --cache-dir /var/tmp/restic --cacert /opt/tools/share/mozilla-rootcerts/cacert.pem -o b2.connections=10 backup /myfiles/.zfs/snapshot/restic

repository 4eab713f opened successfully, password is correct
no parent snapshot found, will read all files
Fatal: unable to save snapshot: NodeFromFileInfo: Listxattr: xattr.list /myfiles/.zfs/snapshot: invalid argument

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Backblaze B2

Expected behavior
-----------------

Continue backup when encountering parental directories without extended attribute support.

Actual behavior
---------------

Restic aborts backup with fatal error.

Steps to reproduce the behavior
-------------------------------

1. Install SmartOS

2. Create a zfs file system and snapshot it.
~~~
zfs create -o xattr=on -o mountpoint=/myfiles zones/myfiles
zfs snapshot zones/myfiles@restic
~~~

3. Confirm xattr and snapshot existence
~~~
zfs get xattr zones/myfiles
zfs list -t snapshot zones/myfiles@restic
~~~

4. Perform backup and observe the fatal error
~~~
/opt/local/bin/restic --cache-dir /var/tmp/restic --cacert /opt/tools/share/mozilla-rootcerts/cacert.pem -o b2.connections=10 backup /myfiles/.zfs/snapshot/restic/

repository 4eab713f opened successfully, password is correct
no parent snapshot found, will read all files
Fatal: unable to save snapshot: NodeFromFileInfo: Listxattr: xattr.list /myfiles/.zfs/snapshot: invalid argument
~~~

Do you have any idea what may have caused this?
-----------------------------------------------

Per https://github.com/openzfs/zfs/issues/7253 .zfs and .zfs/snapshot directories do not support xattrs
Maybe related to https://github.com/restic/restic/pull/3034

Do you have an idea how to solve the issue?
-------------------------------------------

Happy to help test solutions.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

It made me fatally sad ;-)

Overall, been a solid robust solution and its broad platform support is a real bonus over alternative offerings. Not everyone lives in the Linux / Windows world and having a solution prepared to support FreeBSD, Solaris etc is very nice.

**Comment by greatroar** [comment]
On Solaris, getting the xattrs involves an openat call with O_XATTR. That fails with EINVAL instead of ENOTSUP according to [Illumos's manpage](https://illumos.org/man/2/openat). We'd have to handle that, as well as ENOTSUP, or we'd have to convince the pkg/xattr folks to change the error for us.

Since this is SmartOS, could you try the Linux binary as a workaround?

**Comment by creaky** [comment]
Linux binary operates within a LX branded zone. Created such a zone and exposed a ZFS file system through LOFS mount. This permitted restic to successfully backup the snapshot available under the LOFS mount.

However, some of the limitations include very broken snapshot permission issues, and ACLs when accessing snapshots under LX zones making this a fragile workaround at best and highly likely to fail in the presence of many snapshots.

I repeated the effort in a native zone and it failed with the same originally reported error.

For those who wish to duplicate, the way to achieve the LX workaround is:

* Create a ZFS file system with a snapshot in the global zone.
~~~
zfs create -o xattr=on -o mountpoint=/myfiles zones/myfiles
zfs snapshot zones/myfiles@restic
~~~
* Create linux branded zone running CentOS Stream release 8.
~~~
{
	"brand": "lx",
	"kernel_version": "4.3.0",
	"image_uuid": "ef0af4a0-0a05-4852-b891-53430706bc5b",
	"alias": "restic",
	"hostname": "restic",
	"max_physical_memory": 1024,
	"quota": 100,
	"resolvers": ["redacted", "redacted"],
	"nics": [
	{
		"nic_tag": "redacted",
		"ip": "redacted",
		"netmask": "redacted",
		"gateway": "redacted",
		"vlan_id": redacted
	}
	],
        "filesystems": [
        {
                "source": "/myfiles",
                "target": "/source/myfiles",
                "type": "lofs",
                "options": [ "ro" ]
         }
         ]
}
~~~

* Start restic LX zone and do the usual zone updates etc.
* Use dnf to install restic from EPEL, at time of writing being restic 0.13.1 compiled with go1.17.7 on linux/amd64
* Create a new restic repository and perform the backup. Following is a local restic repository example.
~~~
mkdir /restic
restic -r /restic init
restic -r /restic backup /source/myfiles/.zfs/snapshot/restic
~~~
Success:
~~~
repository 18740738 opened successfully, password is correct
created new cache in /root/.cache/restic
no parent snapshot found, will read all files

Files:           2 new,     0 changed,     0 unmodified
Dirs:            4 new,     0 changed,     0 unmodified
Added to the repo: 2.102 KiB

processed 2 files, 36 B in 0:00
snapshot d6a15d73 saved
~~~
